pkgs/tools/security/nwaku/default.nix

```
@@ -0,0 +1,89 @@
+{
+  stdenv, fetchurl, fetchgit, lib, nim, which, writeScriptBin,
+  # Options: wakunode1, wakunode2, wakubridge
+  makeTargets ? [ "wakunode2" ],
+  # WARNING: CPU optmizations that make binary not portable.
+  nativeBuild ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nwaku";
+  version = "v0.15.0";
+  commit = "e12b7cb4";
+  name = "${pname}-${version}-${commit}";
+
+  src = fetchgit {
+    url = "https://github.com/waku-org/nwaku.git";
+    rev = version;
+    sha256 = "sha256-tI3EU4PlUGHPqH0PyptJoX3TcgvQxYeIjCMIITYqjS0=";
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    # Fix import issues: 'handlers.nim:11:27 Error: cannot open file: ../../../../waku/v2/node/waku_node'
+    (fetchurl {
+      url = "https://github.com/waku-org/nwaku/commit/0bf8d573a7f85e67c34d21176edfbe9aa893de11.patch";
+      sha256 = "sha256-gU4bdoezZlqhPAa0ItUB1+CuBHraJWFzwnLwQmDB6Rk=";
+    })
+  ];
+
+  buildInputs = let
+    # Fix for Nim compiler calling 'git rev-parse' and 'lsb_release'.
+    fakeGit = writeScriptBin "git" "echo $commit";
+    fakeLsbRelease = writeScriptBin "lsb_release" "echo nix";
+  in [ fakeGit fakeLsbRelease which nim ];
+
+  enableParallelBuilding = true;
+
+  # Avoid make calling 'git describe'.
+  GIT_VERSION = version;
+
+  NIMFLAGS = lib.optionalString (!nativeBuild) " -d:disableMarchNative";
+
+  # Use available Nim and avoid recompiling Nim for every build.
+  makeFlags = makeTargets ++ [ "USE_SYSTEM_NIM=1" ];
+
+  # Generate vendor/.nimble contents with correct paths.
+  configurePhase = ''
+    runHook preConfigure
+    export HOME=$PWD
+    export PWD_CMD=$(which pwd)
+    export EXCLUDED_NIM_PACKAGES=""
+    export NIMBLE_LINK_SCRIPT=$PWD/vendor/nimbus-build-system/scripts/create_nimble_link.sh
+    export NIMBLE_DIR=$PWD/vendor/.nimble
+    patchShebangs $PWD/vendor/nimbus-build-system/scripts > /dev/null
+    for dep_dir in $(find vendor -type d -maxdepth 1); do
+        pushd "$dep_dir" >/dev/null
+        $NIMBLE_LINK_SCRIPT "$dep_dir"
+        popd >/dev/null
```

```suggestion
        pushd "$dep_dir"
        $NIMBLE_LINK_SCRIPT "$dep_dir"
        popd
```
I think some debugging help here does not hort

---

Sure.